gninasrc/lib/cnn_scorer.cpp

```diff
@@ -295,6 +301,151 @@ void CNNScorer::get_net_output(Dtype& score, Dtype& aff, Dtype& loss) {
 
   loss = lossblob->cpu_data()[0];
 }
+
+// Extract ligand atoms and coordinates
+void CNNScorer::setLigand(const model& m){
+
+  sz n = m.get_movable_atoms().size();
+
+  // Preallocation (on first call)
+  ligand_coords.reserve(n);
```

Comment:
Don't you need to clear these so they have size 0?

Reply:
I was working under the assumption that atom types don't change so that `setLigand` and `setReceptor` are used both to initialize the atom types and coordinates (first call of `CNNScorer::score`) and update the coordinates (on subsequent calls of `CNNScorer::score`). 

However I suspect this assumption is wrong in general (otherwise it would make sense to link the `model` to `CNNScorer` at construction) and this is why I was asking below in the PR discussion about this. If the assumption is wrong, then I should definitely clear the atom types as well (causing small loss in efficiency in the cases where `CNNScorer::score` is called multiple times with the same model).

Reply:
I think it's more of the programming idiom.  I don't expect a call to reserve unless the vector is about to be appended to, but this is always called.  I would move the reserve under the size == 0 check and make the else to `size == 0` be `size == n` with a final else an assert

Reply:
Oh, I see. I agree it's confusing. I'll change it the way you suggest!